ossdbtoolsservice/schema/schema_editor_service.py

```diff
@@ -0,0 +1,186 @@
+        # Make sure we have the appropriate session params
+        validate.is_not_none("params", params)
+
+        params = get_connection_details_with_defaults(params)
+
+        # Generate the session ID and create/store the session
+        session_id = schema_utils.generate_session_uri(params)
+        owner_uri = ObjectExplorerService._generate_session_uri(params)
+
+        if self._logger:
+            self._logger.info(f"   - Session ID: {session_id}")
+
+        # Add the session to session map in a lock to
+        # prevent race conditions between check and add
+        with self._session_lock:
+            if session_id in self._session_map:
+                message = f"Schema Designer session for {session_id} already exists."
+                # If session already exists, get it and respond with it
+                if self.service_provider.logger is not None:
+                    self.service_provider.logger.info(message)
+                request_context.send_error(message)
+            else:
+                # If session doesn't exist, create a new one
+                session = SchemaEditorSession(session_id, owner_uri)
+                self._session_map[session_id] = session
+                session.initialize(request_context)
+
+        if self._logger:
+            self._logger.info(f"   - Session created: {session_id}")
+
+        return
+
+    
+    def _handle_get_schema_model_request(
+        self, request_context: RequestContext, params: SessionIdContainer
+    ) -> None:
+        """Handle a get schema model request"""
+
+        try:
+            validate.is_not_none("session_id", params)
+            session_id = params.session_id
+            with self._session_lock:
+                assert(session_id in self._session_map)
+                session = self._session_map[session_id]
+                assert(session.init_task is None)
+            connection = self._conn_service.get_connection(
+                session.owner_uri, ConnectionType.OBJECT_EXLPORER
+            )
+            if not connection:
+                request_context.send_error(f"No connection available for {session.owner_uri}")
+                return    
+            session.get_schema_model(request_context, connection)
+        except Exception as e:
+            message = f"Failed to get schema model: {str(e)}"
+            if self.service_provider.logger is not None:
+                self.service_provider.logger.error(message)
+            request_context.send_error(message)
+            return
+        return
+
+    def _handle_close_session_request(
+        self, request_context: RequestContext, params: SessionIdContainer
+    ) -> None:
+        try:
+            validate.is_not_none("session_id", params)
+            session_id = params.session_id
+            with self._session_lock:
+                assert(session_id in self._session_map)
+                session = self._session_map.pop(session_id)
+                session.close_session()
```

Comment:
If the schema session ID is also associated with the connection pool via ConnectionService.connect, here you will need to call a ConnectionService.disconnect to disassociate it properly.